### PR TITLE
Chore: Hooks app/views/DefaultBrowserView

### PR DESCRIPTION
**Pull request by reinaldonetof**
## Proposed changes

Migrate DefaultBrowserView to hooks

## How to test or reproduce

- After logged
- Join a room, click on any link and see how the web page is opened
- Then, go to Settings and click on `Default browser`
- Change to another type of browser, back to that room in step 2 and open again the link

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] Improvement (non-breaking change which improves a current function)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Documentation update (if none of the other choices apply)

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/RocketChat/Rocket.Chat/blob/develop/.github/CONTRIBUTING.md#contributing-to-rocketchat) doc
- [x] I have signed the [CLA](https://cla-assistant.io/RocketChat/Rocket.Chat.ReactNative)
- [x] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works (if applicable)
- [ ] I have added necessary documentation (if applicable)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: app/views/DefaultBrowserView.tsx [hunk 1]
```diff
@@ -48,106 +46,95 @@ const BROWSERS: IBrowsersValues[] = [
...
-	browser: string | null;
-	supported: IBrowsersValues[];
+interface IRenderItem extends IBrowsersValues {
+	browser: string;
+	theme: TSupportedThemes;
+	changeDefaultBrowser: (newBrowser: TValue) => void;
 }
 
-type IDefaultBrowserViewProps = IBaseScreen<SettingsStackParamList, 'DefaultBrowserView'>;
+const RenderItem = memo(({ title, value, browser, changeDefaultBrowser, theme }: IRenderItem) => {
```
Comment (diegolmello) [inline app/views/DefaultBrowserView.tsx:55]: This is anti pattern.
`render*` (with lowercase `r`) are functions to render stuff.
`Item` (with uppercase `I`) is a react component.
`RenderItem` is a mix of both, that's why you shouldn't do it.

It should be `Item` and on a separate file.

File: app/views/DefaultBrowserView.tsx [hunk 2]
```diff
@@ -48,106 +46,95 @@ const BROWSERS: IBrowsersValues[] = [
...
-		}
-	}
+const DefaultBrowserView = () => {
+	const [browser, setBrowser] = useState<string>('');
+	const [supported, setSupported] = useState<IBrowsersValues[]>([]);
 
-	componentDidMount() {
-		const browser = UserPreferences.getString(DEFAULT_BROWSER_KEY);
-		this.setState({ browser });
-	}
```
Comment (diegolmello) [inline app/views/DefaultBrowserView.tsx:77]: I think you've removed this whole logic here 🤔

**Review by diegolmello — CHANGES_REQUESTED** [review]

File: app/views/DefaultBrowserView/index.tsx [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+		<SafeAreaView testID='default-browser-view'>
+			<StatusBar />
+			<FlatList
+				data={DEFAULT_BROWSERS.concat(supported)}
+				keyExtractor={item => item.value}
+				contentContainerStyle={List.styles.contentContainerStyleFlatList}
+				renderItem={({ item }) => (
+					<Item
+						browser={browser}
+						theme={theme}
```
Comment (GleidsonDaniel) [inline app/views/DefaultBrowserView/index.tsx:99]: We don't pass the theme via prop drilling anymore.

File: app/views/DefaultBrowserView/Item.tsx [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+import * as List from '../../containers/List';
+import { IBrowsersValues, TValue } from '.';
+
+interface IRenderItem extends IBrowsersValues {
+	browser: string | null;
+	theme: TSupportedThemes;
+	changeDefaultBrowser: (newBrowser: TValue) => void;
+}
+
+const Item = memo(({ title, value, browser, changeDefaultBrowser, theme }: IRenderItem) => {
```
Comment (GleidsonDaniel) [inline app/views/DefaultBrowserView/Item.tsx:15]: Is this really necessary?
Considering the amount of options available, I don't see the need to use memo in this component.

It's not a matter of being right or wrong, it's a matter of making the simple complex.

File: app/views/DefaultBrowserView/index.tsx [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+		navigation.setOptions({
+			title: I18n.t('Close_Chat')
+		});
+	}, [navigation]);
+
+	useEffect(() => {
+		const getBrowser = UserPreferences.getString(DEFAULT_BROWSER_KEY);
+		setBrowser(getBrowser);
+
+		if (isIOS) {
```
Comment (GleidsonDaniel) [inline app/views/DefaultBrowserView/index.tsx:66]: The old code doesn't have this logic, are you sure it's just for ios?

**Review by GleidsonDaniel — CHANGES_REQUESTED** [review]

File: app/views/DefaultBrowserView/index.tsx (hunk 5 above)
Comment (reinaldonetof) [reply app/views/DefaultBrowserView/index.tsx:66]: Actually, there is.

https://github.com/RocketChat/Rocket.Chat.ReactNative/pull/4424/files#diff-4161ff0a346cc7ff7d7339326a01e6ca3a700901f02ef5160730cb90013625a4L69-L71

File: app/views/DefaultBrowserView/Item.tsx [hunk 6]
```diff
@@ -0,0 +1,35 @@
+import React from 'react';
+
+import I18n from '../../i18n';
+import { TSupportedThemes, useTheme } from '../../theme';
+import * as List from '../../containers/List';
+import { IBrowsersValues, TValue } from '.';
+
+interface IRenderItem extends IBrowsersValues {
+	browser: string | null;
+	theme: TSupportedThemes;
```
Comment (GleidsonDaniel) [inline app/views/DefaultBrowserView/Item.tsx:10]: 👀

File: app/views/DefaultBrowserView/index.tsx (hunk 3 above)
Comment (GleidsonDaniel) [inline app/views/DefaultBrowserView/index.tsx:99]: 👀

**Review by GleidsonDaniel — CHANGES_REQUESTED** [review]

**Review by GleidsonDaniel — APPROVED** [review]

**Pull request merged**